torch/testing/asserts.py

```diff
@@ -0,0 +1,389 @@
+import functools
+import inspect
+import sys
+import warnings
+from typing import Any, Callable, Optional, Tuple
+
+import torch
+from torch.testing._internal.common_utils import get_comparison_dtype as _get_comparison_dtype, TestCase as _TestCase
+
+__all__ = ["assert_tensors_equal", "assert_tensors_allclose"]
+
+
+# The module 'pytest' will be imported if the 'pytest' runner is used. This will only give false-positives in case
+# the test directly or indirectly imports 'pytest', but is run by another runner such as 'unittest'.
+_RUN_BY_PYTEST = "pytest" in sys.modules
+
+
+# The Usage(Error|Warning) should be raised in case the test function is not used correctly. With this the user is able
+# to differentiate between a test failure (there is a bug in the tested code) and a test error (there is a bug in the
+# test). If pytest is the test runner, we use the built-in UsageError instead our custom one.
+if _RUN_BY_PYTEST:
+    import pytest
+
+    UsageError = pytest.UsageError
+else:
+    class UsageError(Exception):
+        pass
+
+
+class UsageWarning(Warning):
+    pass
+
+
+def _get_default_rtol_and_atol(a: torch.Tensor, b: torch.Tensor) -> Tuple[float, float]:
+    try:
+        return _TestCase()._getDefaultRtolAndAtol(a.dtype, b.dtype)
```

Comment:
Constructing a new TestCase every time we want to acquire the default rtol and atol seems like a pain. I suggest making _getDefaultRtolAndAtol and the data structure it uses not a method of TestCase and importing the function from common_utils.py.